.github/workflows/test.yml

```diff
@@ -5,22 +5,29 @@ jobs:
   ruby:
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
-        ruby: [ '2.3', '2.4', '2.5' ]
+        ruby: [ '2.6', '2.7', '3.0' ]
+        gemfile: [ 'rails_5.2', 'rails_6.1' ]
+        exclude:
+          - ruby: '3.0'
+            gemfile: 'rails_5.2'
+    env:
+      BUNDLE_GEMFILE: gemfiles/${{ matrix.gemfile }}.gemfile
```

Comment:
Explicitly set the gem file to use.

Previously, each Ruby build step would `bundle install` and run the test suite against multiple gem files. This obscured the test result and slowed the build.

![image](https://user-images.githubusercontent.com/497874/120779755-bc392900-c56a-11eb-988c-b0bb07c1a96f.png)


Now each Ruby build step will `bundle install` and run the test suite against a single gem file. Running the build this way, we can take advantage of the caching provided by the `ruby/setup-ruby` plugin.

![image](https://user-images.githubusercontent.com/497874/120780326-57ca9980-c56b-11eb-8efb-c246402a6194.png)



The CI pipeline is blazingly fast and informative now!